aperturedb/Utils.py

```diff
@@ -617,45 +617,30 @@ def remove_all_objects(self):
 
         cmd = {"constraints": {"_uniqueid": ["!=", "0.0.0"]}}
 
-        queries = [
-            [{"DeleteEntity": cmd}],
-            [{"DeleteImage": cmd}],
-            [{"DeleteVideo": cmd}],
-            [{"DeleteBlob": cmd}],
+        transaction = [
+            {"DeleteEntity": cmd},
+            {"DeleteImage": cmd},
+            {"DeleteVideo": cmd},
+            {"DeleteBlob": cmd},
             # DeleteDescriptorSet also deletes the descriptors
-            [{"DeleteDescriptorSet": cmd}],
+            {"DeleteDescriptorSet": cmd},
 
             # All the following should be deleted automatically once the
             # related objects are deleted.
             # We keep them here until ApertureDB fully implements this.
-            [{"DeleteBoundingBox": cmd}],
-            [{"DeletePolygon": cmd}],
-            [{"DeleteFrame": cmd}],
+            {"DeleteBoundingBox": cmd},
+            {"DeletePolygon": cmd},
+            {"DeleteFrame": cmd},
+            {"GetSchema": {"refresh": True}}
         ]
 
         try:
-            try:
-                for q in queries:
```

Comment:
the reason why this was a loop is because it was only practical to remove things on smaller transactions rather than 1 giant transaction. I think that is still the case, but we can try. @venkat-murty any thoughts?

In any case, I think this change is good because this is the end goal: a simple queries that removes everything. 
If we see that it is not practical to remove everything on a single transaction, we can implement something that deletes 10000 things at a time, check the schema, moves on the next 10000s, and iterate.

@bovlb

Reply:
If we have enough memory then it should not be a problem. If we are deleting n objects then it will create in-memory n objects + an object for every connection + an object for all the connected nodes.

Reply:
I would suggest limiting by some using limit = x, safer.

Reply:
Another problem, a large delete can take a long time (so timeouts are possible) and you have no idea if the system is progressing.

In addition, we (should) would add time limits on queries for system stability in near future.

Reply:
> I would suggest limiting by some using limit = x, safer.

Yes, loops with a limit would be best, it can be in a followup PR too.